### PR TITLE
docs: Support dark mode in new docs site

### PR DESCRIPTION
**Pull request by jdeichert**
## Motivations

I quickly hacked in support for dark mode in the new docs site!

NOTE: The video below is a little outdated, the button is now BOTTOM RIGHT instead of top right.

https://github.com/user-attachments/assets/d9548c37-a521-4aa6-a53a-439b7ac13701

### Added

- Added dark mode support to new docs site

## Testing

1. Run `npm run dev` so the `dark.mode.css` file gets included in the public editor bundle
2. Load the site
3. See the theme toggle button top right, click it to toggle the theme
4. Verify the code preview renders using the correct theme
5. Verify the theme can be overridden using the url param `?theme=dark`
6. Verify a page reload restores the last used theme (from local storage)

Changes can be [tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

### CODE REVIEW

File: packages/site/src/components/ContentCard.tsx [hunk 1]
```diff
@@ -26,10 +31,19 @@ export const ContentCard = ({
...
       {!component ? (
-        <img
-          style={{ width: "100%", mixBlendMode: "multiply" }}
-          src={imageURL}
-        />
+        <AtlantisThemeContextProvider dangerouslyOverrideTheme="light">
+          <div
+            style={{
+              background: "var(--color-surface)",
+              borderRadius: "var(--radius-base) var(--radius-base) 0 0",
```
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/site/src/components/ContentCard.tsx:38]: Not something we need to address here / might take a bit of thinking, but I'm _pretttttty_ sure we should add `overflow: hidden` to Card 🤔

File: packages/site/src/components/ContentCard.tsx [hunk 2]
```diff
@@ -27,10 +34,23 @@ export const ContentCard = ({
       }}
     >
       {!component ? (
-        <img
-          style={{ width: "100%", mixBlendMode: "multiply" }}
-          src={imageURL}
-        />
+        <AtlantisThemeContextProvider dangerouslyOverrideTheme="light">
```
Comment (scotttjob) [inline packages/site/src/components/ContentCard.tsx:37]: TASTY DELICIOUS DOG FOOD. Love it.

File: packages/site/src/components/ContentCard.tsx [hunk 3]
```diff
@@ -27,10 +34,23 @@ export const ContentCard = ({
...
+          <div
+            style={
+              imageURL && theme === "dark"
+                ? {
```
Comment (scotttjob) [inline packages/site/src/components/ContentCard.tsx:41]: It's legit a practical example of how to use our stuff just right here in the code. Just wonderful.

File: packages/site/src/components/ContentCard.tsx [hunk 4]
```diff
@@ -26,10 +31,19 @@ export const ContentCard = ({
...
+              background: "var(--color-surface)",
+              borderRadius: "var(--radius-base) var(--radius-base) 0 0",
+            }}
+          >
+            <img
+              style={{ width: "100%", mixBlendMode: "multiply" }}
+              src={imageURL}
+            />
+          </div>
+        </AtlantisThemeContextProvider>
```
[the remark this answers is not in the document]
Comment (scotttjob) [reply packages/site/src/components/ContentCard.tsx:53]: Something we can probably tweak in an upcoming version!

File: packages/site/src/components/ToggleThemeButton.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        top: "var(--space-base)",
+        right: "var(--space-base)",
+      }}
+    >
+      <Button
+        type="secondary"
+        label={isDark ? "☀️" : "🌕"}
+        onClick={handleClick}
+      />
+    </div>
```
[the remark this answers is not in the document]
Comment (scotttjob) [reply packages/site/src/components/ToggleThemeButton.tsx:32]: Awesome! We can move it if need be. :)

File: packages/site/src/main.tsx [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
+import { AtlantisThemeContextProvider } from "@jobber/components";
 import { Layout } from "./layout/Layout";
 import { AtlantisPreviewEditorProvider } from "./providers/AtlantisPreviewEditorProvider";
 import { AtlantisSiteProvider } from "./providers/AtlantisSiteProvider";
+import { initAtlantisTheme } from "./utils/theme";
 
 const urlParams = new URLSearchParams(window.location.search);
 const minimalMode = urlParams.has("minimal");
 
+initAtlantisTheme(urlParams.get("theme"));
```
[the remark this answers is not in the document]
Comment (scotttjob) [reply packages/site/src/main.tsx:17]: Honestly feels like we could just point our documentation at our source code for a practical example of how all this works.

File: packages/site/src/providers/AtlantisPreviewEditorProvider.tsx [hunk 7]
```diff
@@ -186,6 +192,9 @@ export const AtlantisPreviewEditorProvider = ({
 
                 ${transpiledCode}
 
+            function RootWrapper() {
```
Comment (scotttjob) [inline packages/site/src/providers/AtlantisPreviewEditorProvider.tsx:195]: Awesome that you figured this out on your own. It's just code! I think this file can be cleaned up and improved, but overall I think we're onto something here! :)

File: packages/site/src/providers/AtlantisPreviewEditorProvider.tsx [hunk 8]
```diff
@@ -210,6 +216,13 @@ export const AtlantisPreviewEditorProvider = ({
     [iframe],
   );
 
+  useEffect(() => {
+    if (iframe.current) {
+      const iframeWindow = iframe.current.contentWindow;
+      iframeWindow?.postMessage({ type: "updateTheme", theme }, "*");
+    }
+  }, [theme]);
```
[the remark this answers is not in the document]
Comment (scotttjob) [reply packages/site/src/providers/AtlantisPreviewEditorProvider.tsx:228]: Absolutely fantastic. Huge thumbs up.

File: packages/site/src/providers/AtlantisPreviewEditorProvider.tsx [hunk 9]
```diff
@@ -300,10 +316,9 @@ export const AtlantisPreviewEditor = () => {
   );
 };
 
-const skeletonHTML = `
-
+const skeletonHTML = (theme: Theme) => `
 <!DOCTYPE html>
-<html>
+<html data-theme="${theme}">
```
Comment (scotttjob) [inline packages/site/src/providers/AtlantisPreviewEditorProvider.tsx:321]: 😄

File: packages/site/src/utils/theme.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import { Theme } from "@jobber/components";
+
+const FALLBACK_THEME: Theme = "light";
```
Comment (scotttjob) [inline packages/site/src/utils/theme.ts:3]: Are these helpers that we should be shipping?! Have we just improved the design system like 3 times as a result of this PR??

**Review by scotttjob — APPROVED** [review]
LGTM! I even ran this code locally which I don't normally do. I wanted to see it all in action. It's amazing! Bravo!

File: packages/site/src/utils/theme.ts (hunk 10 above)
Comment (scotttjob) [reply packages/site/src/utils/theme.ts:3]: Fair!

**Pull request merged**